### PR TITLE
RDoc-1317 TestDrive

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: Documentation/4.1/Raven.Documentation.Pages/start/.docs.json [hunk 1]
```diff
@@ -36,13 +36,10 @@
     {
         "Path": "test-driver.markdown",
         "Name": "Unit Tests",
-        "Mappings": [
-            {
-                "Version": 3.0,
-                "Key": "server/troubleshooting/sending-support-ticket"
-            }
-        ],
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/.docs.json]: Why did you remove the mapping to 3.0?

File: Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 2]
```diff
@@ -0,0 +1,108 @@
+﻿# Getting Started : Writing your Unit Test using TestDriver
+
+In this section we will explain how to use [RavenDB.TestDriver](https://www.nuget.org/packages/RavenDB.TestDriver/) in order to write unit tests for working with RavenDB.
+TestDriver now uses [RavenEmbedded](../server/Embedded) to run the server, in this way we can just start using it without preparations.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: TestDriver now uses -> TestDriver uses

Comment (arekpalinski) [reply Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: RavenEmbedded -> RavenDB.Embedded - to be consistent with the package name or just "embedded server"

Comment (arekpalinski) [reply Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: "in this way we can just start using it without preparations." - I don't think it's necessary to stress it out, that's how it just works in 4.1

File: Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Pre-Initializing the IDocumentStore allows you to mutate the conventions used by the document store.
+
+### Example
+
+{CODE test_driver_PreInitialize@Start\RavenDBTestDriver.cs /}
+
+{PANEL/}
+
+{PANEL:UnitTest}
+I'm using [xunit](https://www.nuget.org/packages/xunit/) for my test framework in the below example.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: "I'm using" - we typically use "Let's use ..." or "We'll be using .."

File: Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+Note that the test itself is meant to show diffrent capabilities of the test driver and is not meant to be the most efficient.
+The example below depends on the `TestDocumentByName` index and `TestDocument` class that can be seen in the [full example](../start/test-driver#complete-example)
+
+### Example
+
+{CODE test_driver_MyFirstTest@Start\RavenDBTestDriver.cs /}
+
+In the test we get an IDocumentStore to our test database, deploy an index and insert two documents into it.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: Regarding above comment - here we use "we get"

File: Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+{CODE test_driver_MyFirstTest@Start\RavenDBTestDriver.cs /}
+
+In the test we get an IDocumentStore to our test database, deploy an index and insert two documents into it. 
+We then wait for the indexing to complete and launch the Studio so we can verify the documents and index are deployed (we can remove this line once the test is working).
+At the end of the test we query for TestDocument where their name contains the world 'hello' and assert that we have only one such document.
+
+{PANEL/}
+
+{PANEL: ConfigureServer}
+`ConfigureServer` allows you to be more in control on your server.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: Maybe let's change `ConfigureServer` to `ConfigureServer()` or `ConfigureServer`  method - to indicate that's the method of the base test class

File: Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+In the test we get an IDocumentStore to our test database, deploy an index and insert two documents into it. 
+We then wait for the indexing to complete and launch the Studio so we can verify the documents and index are deployed (we can remove this line once the test is working).
+At the end of the test we query for TestDocument where their name contains the world 'hello' and assert that we have only one such document.
+
+{PANEL/}
+
+{PANEL: ConfigureServer}
+`ConfigureServer` allows you to be more in control on your server. 
+You can use it with `ServerTestOptions` to change the path to you server dll or to specify where your RavenDB data is stored, security, etc.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: to you server dll -> to the Raven server binaries?

File: Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+{INFO:ServerTestOptions}
+
+`ServerTestOptions` inherits from `ServerOptions` in that way you can be more in control on how the embedded server going to run.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: `ServerOptions` let's put a link here and drop below "For more inforamtion go ..." sentence

Comment (arekpalinski) [reply Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: to run. -> to run (as the sentence is continued below)

File: Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+{PANEL/}
+
+
+{PANEL:Continuous Integration Servers}
+
+Best practice is to use a CI/CD server to help automate the testing and deployment of your new code. 
+Popular CI/CD products are [AppVeyor](https://www.appveyor.com/) or [Visual Studio Team Services (aka. VSTS)](https://www.visualstudio.com/team-services/). Some customization is required for any
+CI/CD product you use, because you will need to manually download the RavenDb Server _before_ any tests are kicked off. Remember, the Test Driver
+requires a `path location` for a `Raven.Server.exe` or `Raven.Server.dll` to be located, where the path on your CI/CD server
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: What's "path location" ? Is it sill relevant since we use embedded? Is it 'ServerTestOptions.ServerDirectory' ?

File: Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+Popular CI/CD products are [AppVeyor](https://www.appveyor.com/) or [Visual Studio Team Services (aka. VSTS)](https://www.visualstudio.com/team-services/). Some customization is required for any
+CI/CD product you use, because you will need to manually download the RavenDb Server _before_ any tests are kicked off. Remember, the Test Driver
+requires a `path location` for a `Raven.Server.exe` or `Raven.Server.dll` to be located, where the path on your CI/CD server 
+will most likely be different to the path on your localhost-development machine.
+
+{PANEL/}
+
+## Related articles
+
+- [RavenEmbedded](../server/Embedded)
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/start/test-driver.dotnet.markdown]: Let's use article name here:

`[Running an Embedded Instance](../../server/embedded)`

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+        [Fact]
+        public void TestDriveConfigureServer()
+        {
+            #region test_driver_ConfigureServer
+
+            var testServerOptions = new TestServerOptions
+            {
+                ServerDirectory = "PATH_TO_RAVENDB_SERVER", // If we want to specify ravendb server location
+                DataDirectory = "PATH_TO_RAVENDB_DATADIR", // Specify where ravendb data
```
Comment (arekpalinski) [inline Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs]: Specify where ravendb data -> Specify where ravendb data will be placed / located ?

Comment (arekpalinski) [reply Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs]: I wonder if we should mark here that a user is not required to fill those props. They are to allow to specify only _custom_ paths, right?

**Comment by arekpalinski** [comment]
BTW in the future please apply the changes to a copied article in a separate commit so it's easy to see what exactly has been changed

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs (hunk 10 above)
Comment (IdanHaim) [reply Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/Start/RavenDBTestDriver.cs]: In the docs I wrote to be more in control I will add (Optional)
